samples/Android/app/src/main/java/software/amazon/awssdk/iotsamples/MainActivity.java

```diff
@@ -301,7 +301,8 @@ public void run(){
                     sampleMain.invoke(null, (Object) args);
                 }
             } catch (Exception e){
-                writeToConsole("Exception occurred in run(): " + e.toString() + "\n");
+                writeToConsole("Exception occurred in run(): " + e.toString() +
+                "\nCause: " + e.getCause().toString());
```

Comment:
The error string seems to be removed. Was it because the `getCause()` not supported in 24?

Reply:
Unsure what you mean. The Exception caught here is printed as `e.toString()` and that always simply resulted in "java.lang.reflect.InvocationTargetException". I've added an additional `e.getCause().toString()` to provide some context related to what the underlying error/exception was if available.

Reply:
Ah never mind, my mistake. I was confused with getCause() and the e.toString().